### ISSUE TITLE
Kite | Install fail on Windows10 - 64bits - Intel Core i5 8th Gen

### ISSUE THREAD

**Issue by CaptainKey** [kiteco/plugins#158]
Hi,

I tried to install Kite on my computer but i had this error : 

<img width="307" alt="Kite" src="https://user-images.githubusercontent.com/25344190/70444631-7f39c680-1a9a-11ea-97a9-8bf15e15441e.png">

My Laptop is an XPS Dell 13" with Windows 10 - 64 bits and Intel Core i5 8th Gen (It support AVX).

Is anyone have the same issue ? 

Regards,

**Comment by asteppke** [comment]
Yes. Same issue here, similar system Windows 10, 64 bits, AMD Zen 2600. 

The error message only mentions that Windows defender might slow things down so I disabled that temporarily but the failure remains. The installation seems to almost succeed, the files are in `C:\Program Files\Kite`, the `KiteService.exe` is running but Visual Studio code thinks that the engine is not running so something is still missing.

**Comment by adamsmith** [comment]
Hi all, if you try to run `C:\Program Files\Kite\kited.exe` I'm guessing it won't start?

After trying to run `kited.exe`, is there a file at `%USERPROFILE%\AppData\Local\Kite\logs\client.log`?

If so, could you send it? (If it has anything you'd like to keep private inside you can email it to me at adam@kite.com.)

If that file doesn't exist, then something low level is stopping `kited` from running. The next debugging step would be to run [WinDbg](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/getting-started-with-windbg) which is pretty esoteric. If you're feeling adventurous you could try to see if there's an uncaught exception that's crashing it, or if you're feeling generous you could email me (adam@kite.com) and we could set up a 20 minute remote debugging session where you share your computer over Zoom and I can install and run WinDbg so we can get to the bottom of it. 

Thanks!!!

**Comment by CaptainKey** [comment]
> Hi all, if you try to run `C:\Program Files\Kite\kited.exe` I'm guessing it won't start?
> 
> After trying to run `kited.exe`, is there a file at `%USERPROFILE%\AppData\Local\Kite\logs\client.log`?
> 
> If so, could you send it? (If it has anything you'd like to keep private inside you can email it to me at [adam@kite.com](mailto:adam@kite.com).)
> 
> If that file doesn't exist, then something low level is stopping `kited` from running. The next debugging step would be to run [WinDbg](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/getting-started-with-windbg) which is pretty esoteric. If you're feeling adventurous you could try to see if there's an uncaught exception that's crashing it, or if you're feeling generous you could email me ([adam@kite.com](mailto:adam@kite.com)) and we could set up a 20 minute remote debugging session where you share your computer over Zoom and I can install and run WinDbg so we can get to the bottom of it.
> 
> Thanks!!!

I don't work on windows anymore. However, i remember that the client.log file existed but was empty.

**Comment by berton7** [comment]
Hi, i'm having the exact same problem. In the logs folder there is only an empty copilot.log.
I tried to start manually C:\Program Files\Kite\win-unpacked\Kite.exe, a window with written "Initializing..." appears but it hangs probably waiting for kited.exe to start. Any solution?

**Comment by mohseninima** [comment]
Same problem. Running kited.exe doesn't do anything. In the logs folder there is only an empty copilog.log file.

**Comment by its-dhung** [comment]
Hi @berton7, @aguilera51284, @mohseninima, we'd love to help out. As mentioned above:

> If that file doesn't exist, then something low level is stopping kited from running. The next debugging step would be to run [WinDbg](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/getting-started-with-windbg) which is pretty esoteric. If you're feeling adventurous you could try to see if there's an uncaught exception that's crashing it, or if you're feeling generous you could email me (adam@kite.com) and we could set up a 20 minute remote debugging session where you share your computer over Zoom and I can install and run WinDbg so we can get to the bottom of it.

This is likely a low level issue that we unfortunately haven't been able to reproduce on any of the Windows VM's, so you can either try debugging this with WinDbg, or feel free to email adam@kite.com to setup a debugging session.

**Comment by mohseninima** [comment]
Before I run WinDbg. When running kited.exe, a command prompt window pops up briefly and then disappears. I was able to get a glimpse of what the window was, and it was the cygwin uname.exe. There might be conflict between kite and cygwin.

**Comment by saisandeepvaddi** [comment]
I did not find log file. 
Here's WinDbg output when ran on kited.exe.

[dbg.txt](https://github.com/kiteco/issue-tracker/files/4636663/dbg.txt)

**Comment by adamsmith** [comment]
@mohseninima @saisandeepvaddi thank you for the WinDbg logs!

These are exactly what we need except for one thing: the "Exception Analysis" is telling us about the "Break instruction exception".

What's happening here is that WinDbg gives you a breakpoint right when kited.exe is starting to execute, in case you wanted to set extra breakpoints or do other things before kited.exe starts to run.

When you're at this initial breakpoint instead of running `!analyze -v` just run the `g` command (no `!` or anything — just `g` followed by Enter). This tells WinDbg to start execution of kited.exe.

After you run the `g` command hopefully you see another exception and at that point running `!analyze -v` and sending this log would be awesome! Thank you!

**Comment by mohseninima** [comment]
Woops, it's been a long time since I have used WinDbg. Here is the updated log. I just get a terminated process.

[debug5-20-20.txt](https://github.com/kiteco/issue-tracker/files/4658709/debug5-20-20.txt)

**Comment by saisandeepvaddi** [comment]
Updated one

[dbg-2.txt](https://github.com/kiteco/issue-tracker/files/4658783/dbg-2.txt)

**Comment by adamsmith** [comment]
Awesome thank you @mohseninima and @saisandeepvaddi!

Indeed it looks like kited.exe is just terminating itself.

It may be helpful to get the following two pieces of added info:

### Enable logger

_Before_ typing the first `g`+Enter, type `!logexts.loge`+Enter. This should load the [logger](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/-logexts-loge). I haven't tested this recently so I'm not sure if you need to install this module but according to [the docs](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/logger-and-logviewer) it should be included with the Debugging Tools for Windows, so it should "just work".

After the debug session is over there will be a directory created on your desktop with the log of execution. If so this may help us figure out what's happening.


### Stack trace for the call to `ntdll!NtTerminateProcess`

When you get to `ntdll!NtTerminateProcess+0x14:` could you capture a stack trace? Basically instead of `!analyze -v` just run `kP` (Enter).


Thanks again!

**Comment by mohseninima** [comment]
The !logexts.loge command wasn't saving any logs for some reason so I ran kited.exe through the standalone logger executable. Here are the outputs from WinDbg and the logger.
[debug.txt](https://github.com/kiteco/issue-tracker/files/4659273/debug.txt)
[logger.txt](https://github.com/kiteco/issue-tracker/files/4659274/logger.txt)

**Comment by adamsmith** [comment]
Thanks @mohseninima! Could you let us know which version of Kite you have installed? You can find this by opening `C:\Program Files\Kite`, right-clicking on `kited.exe`, go to Properties, and then Details. You should see a version string along the lines of `1.___.___.___`. The last number will be a small integer and the middle two numbers will represent a date.

Thanks again!

**Comment by mohseninima** [comment]
The version number is 1.2020.513.0.

**Comment by petercr** [comment]
Hello I'm having the same problem as well ☹️ Here's a copy of my [debug.txt](https://github.com/kiteco/issue-tracker/files/4665615/debug.txt) 
It includes the "Break instruction exception - code", so hopefully that helps 😃 

Here's a screenshot with some extra info if that helps: ⬇️ 
![WinDbg_Pic](https://user-images.githubusercontent.com/26460352/82622350-8ddceb80-9bab-11ea-8325-43a66444e542.png)

**Comment by berton7** [comment]
Hey, so for other reasons I had to format my computer, so I tried again to install kite and this time it worked, installing both the vs code and sublime. So this confirms that it's actually a software problem, and not a hardware one I guess.

**Comment by petercr** [comment]
@berton7 Nice thanks for letting us know. 👍 I have a hunch that it may have to do with Virtualization in Windows. 

Did you have Android Studio or any other program that uses a VM?

**Comment by berton7** [comment]
Yes I had android studio installed. If kite uses virtualization it might actually be a problem; I had A LOT of problems trying to run 64 bits vms, from what I understood because hyper v was always enabled, so I had to create another entry in the windows bootloader to disable hyperv entirely and get my 64 bits vms working.

**Comment by petercr** [comment]
I went and tried turning off all of the Virtualization features in Windows. And then tried to re-install Kite. Unfortunately it _didn't work._ The only thing I did not try was disabling Virtualization on the BIOS.  
I guess I could try that 🤷‍♂️

**Comment by adamsmith** [comment]
Hi all, we have done some debugging on this. An updated request: 

Could you try installing the latest version of Kite to see if you can reproduce this on it?

If you reproduce the error, 

(1) please send the Kite client log which can be found at `%USERPROFILE%\AppData\Local\Kite\logs\client.log`.
(2) try running `C:\Program Files\Kite\kited.exe` from the command line to see what output is printed to stdout/stderr, if any.

Thanks!

**Comment by sahsz123** [comment]
kited.exe is 0 bytes on my side, the Kite installation is failing the same way, I ran it with Admin privilegies and even removed my antivirus - no effect. The only log in: %USERPROFILE%\AppData\Local\Kite\logs is copilot.log with 2 lines in it:
2020-07-11T06:06:16.805Z [copilot] error: unhandled error spawn UNKNOWN
2020-07-11T06:06:26.731Z [copilot] error: unhandled error spawn UNKNOWN

What I noticed is that Kite.exe is basically dying after some time during the installation. C:\Program Files\Kite I have win-unpacked\ where the contents seemed to be extracted.

**Comment by evilnguyen** [comment]
Hi @adamsmith , I have used WinDbg on Kited.exe, and this is the output of WinDbg
[Debug.txt](https://github.com/kiteco/issue-tracker/files/4906451/Debug.txt)

**Comment by petercr** [comment]
So that **still did not work** ☹️ The `client.log` file is exactly zero bytes. So no need to upload that. 

Here is the output from WinDbg.exe ⬇️ 
[Debug.txt](https://github.com/kiteco/issue-tracker/files/4963462/Debug.txt)

And here is another screenshot from the Windows Debugger is that helps at all. 👌 
Personally _I'm all good with trying this anymore_, but I do wish you the best of luck. 💯 
And of course if you have any questions, I will be more than happy to answer them 👍 

![WinDbg Screenshot](https://user-images.githubusercontent.com/26460352/88244948-55b56000-cc63-11ea-82cb-56e9f3451802.png)

**Comment by arnaud-soulie** [comment]
Exactly same issue here. Log file remains empty. Kite.exe from win_unpacked keeps hanging. Nothing happens with kited.exe.
Any update on this ?

**Comment by zhenlingcn** [comment]
I have the same problem, and I hope this problem can be solved as soon as possible!

**Comment by icepole-jon** [comment]
I guess that is some system setting or software  cause that probleam, becouse in my other win10x64 pc work fine, and the other got error.  clean system maybe ok

**Comment by ZhuMengliang** [comment]
The same question also happens！And I even tried to install the old version of Kite  but failed. However,my fellow can successfully install Kite on windows10, that's strange! Hope to solve this problem!

**Comment by gemida** [comment]
[copilot.log](https://github.com/kiteco/issue-tracker/files/5098818/copilot.log)
Have the same issue but in my log isnt empty

**Comment by adamsmith** [comment]
Hi all, how much memory is available on your machines? We suspect this issue may be caused by an "out of memory" condition.

**Comment by Stone-wh** [comment]
Windows 10 encounters the same problem, how to solve it

**Comment by arjunachuravi** [comment]
Hi all,
I also got this issue, but to my surprise it turns out to be a missing dll.
My issue started when I tried to install The Kite software from the setup which is about 300MB. I gave admin privilege for the setup to execute,the setup was getting timeout several times.  After some time,I tried again by installing **VC-_Redist 2015 and 2015 to 2019_** by this link:

[LINK_1](https://download.visualstudio.microsoft.com/download/pr/9fe82b83-f3a1-43f5-8f25-ebe24529854c/B4D433E2F66B30B478C0D080CCD5217CA2A963C16E90CAF10B1E0592B7D8D519/VC_redist.x86.exe)

Then my installation failed with timeout error , by my _program files/kite_ dir was not empty. I ran _kited.exe_ and shower that vcruntime140_1.dll was missing, so i downloaded it from:

[LINK_2](https://download.zip.dll-files.com/2f74f7bbf256d0acb305068a6960ea5a/vcruntime140_1.zip?token=hWKcoZ5HzceZOpqmFEadEA&expires=1599885926
)
and placed it in _c:\windows\system32 folder._
and it worked.

hope it works for everyone

**Comment by Stone-wh** [comment]
Try this method, but it still fails. It should be a program problem.

**Comment by petercr** [comment]
Same as @Stone-wh I tried installing **Kite** after having _pretty much every C++ DLL_ I could find. 
And it did not seem to make any difference for me ☹️ 

I still received the same error message as before, after the installation time's out. 
And you cannot manually start Kite either 

Here's what I have installed for my C++ DLL files on my system:

```
Chocolatey v0.10.15
vcredist-all 1.0.0 [Approved] - Possibly broken
vcredist2005 8.1.0.20160118 [Approved] Downloads cached for licensed users
vcredist2015 14.0.24215.20170201 [Approved]
vcredist2010 10.0.40219.2 [Approved]
vcredist2012 11.0.61031 [Approved] Downloads cached for licensed users
vcredist140 14.26.28720.3 [Approved]
vcredist2013 12.0.40660.20180427 [Approved]
vcredist2008 9.0.30729.6163 [Approved] Downloads cached for licensed users
vcredist2017 14.16.27033 [Approved]
Devbox-VCRedist2010 10.0.40219.20130422

```

**Comment by OSHI7** [comment]
[#427] - related issue

**Comment by OSHI7** [comment]
**Required Information**
To help us triage your issue effectively, please make sure you've completed these steps before opening this issue. Issues with not enough information may be deprioritized or ignored.

- [x] [Read our guide](https://help.kite.com/article/96-contact-us-with-logs) on how to report an issue.
- [x] Include your Kite application logs, _taken while you are experiencing your issue_.
- [x] If reporting a performance problem, include your resource usage logs, _taken while you are experiencing your issue_.
- [x] If reporting an issue with Kite's editor plugin, include which editor you are using and the version, and any accompanying logs from your editor's console.

**Describe the bug**
A clear description of what the bug is. Include steps to reproduce the bug if necessary.

**Screenshots**
If applicable, add screenshots to help explain your problem.
---------------------------------------
Hi,
I'm on Windows 7, trying to install the PyCharm plugin.
I am attaching the following logs:
* [client.log](https://github.com/kiteco/issue-tracker/files/5368840/client.log)
* [copilot.log](https://github.com/kiteco/issue-tracker/files/5368841/copilot.log)
* [idea.log](https://github.com/kiteco/issue-tracker/files/5368926/idea.log)


## Details:
* I install the program, and this graphic stays up for a long time:
![2020-10-13---00-19-29, KiteSetupSplashScreen_](https://user-images.githubusercontent.com/23404030/95815066-ef802c80-0ce9-11eb-829d-5a5f7897241c.png)
* I then get this message:
![2020-10-13---00-11-23, KiteSetupSplashScreen_](https://user-images.githubusercontent.com/23404030/95815123-0757b080-0cea-11eb-88a9-39655da701b0.png)
* I tried disabling my anti-virus and reinstalling while it was off. No joy.
* I can see KiteService is running in services.msc
* I have `kited` installed and when I try to run (with, or without administrator privileges) - nothing seems to happen.
* I get an "unable to connect message" at: http://localhost:46624/clientapi/desktoplogin?d=%2Fsettings%2Facccount
* At: kite://settings/logs , I get a popup that spins endlessly and cannot proceed (per: https://help.kite.com/article/96-contact-us-with-logs)
![image](https://user-images.githubusercontent.com/23404030/95815273-530a5a00-0cea-11eb-8bde-5b3239f32694.png)
* In PyCharm i can see the plugin detects that the "Kite Engine is not running"
![image](https://user-images.githubusercontent.com/23404030/95813860-d32ec080-0ce6-11eb-9d98-6b93f158f73c.png)
* I link to this related issue here: [#158 ](https://github.com/kiteco/issue-tracker/issues/158)
Per link above, I tried running `kited.exe` at the command prompt, but got no output
![image](https://user-images.githubusercontent.com/23404030/95815849-9618fd00-0ceb-11eb-8e07-31011255726f.png)
* [per comment](https://github.com/kiteco/issue-tracker/issues/158#issuecomment-640308189) I went into BIOS and disabled Virtualization and re-installed. No Joy.
* [per comment](https://github.com/kiteco/issue-tracker/issues/158#issuecomment-691257194) Tried re-installing VCruntime redist. Nope.
* related issue [#264]

I would like to use this software, but am really stuck.
Can you suggest anything else to resolve?  Unfortunately I will not be able to run the debugger.
Thank you, Josh

**Comment by petercr** [comment]
You could always try using tabnine instead of Kite in PyCharm. 
I could never get Kite to work on Windows 10 ☹️ 

[Here's the link to their site](https://www.tabnine.com/) there's a similar pricing plan where there is a free tier. 
Good luck 👍 I have no affiliation with tabnine

**Comment by metalogical** [comment]
@OSHI7 thanks for the super detailed report. We're currently investigating this problem again, hopefully we can figure it out this time around.

**Comment by icepole-jon** [comment]
After reinstall win10, install kite first time, it works.

**Comment by pozh** [comment]
> You could always try using tabnine instead of Kite in PyCharm.
> I could never get Kite to work on Windows 10 ☹️
> 
> [Here's the link to their site](https://www.tabnine.com/) there's a similar pricing plan where there is a free tier.
> Good luck 👍 I have no affiliation with tabnine

The best solution till yet, unfortunately

**Comment by OSHI7** [comment]
So.. has there been any movement on this?

**Comment by OSHI7** [comment]
This is now working for me for PyCharm & VS-Code on Windows 7.
Thank you for the follow-up! Working with the tool now!!

**Comment by mpv88** [comment]
hi! experiencing exactly the same problem reported by OSHI7 since 1 month, after 1 year functioning properly the co-pilot is stuck "Inizializing" and the plug-in is not working at all on any of the editors I have installed (vscode, spyder, pycharm). Tried to uninstall/reinstall Kite many times but the bug is still there, are there any official updates or fixes on that?
Thank you very much if you could help

**Comment by pixonte** [comment]
@mpv88 try tabnine, to me it seems to be a nice replacement, maybe even better than kite
